src/agent/rustjail/src/validator.rs

```diff
@@ -31,7 +29,10 @@ fn rootfs(root: &str) -> Result<()> {
     let path = PathBuf::from(root);
     // not absolute path or not exists
     if !path.exists() || !path.is_absolute() {
-        return Err(einval());
+        return Err(anyhow!(
+            "Path from {} does not exist or is not absolute",
```

Comment:
I'd use `{:?}` for all paths to ensure they are quoted (so you can see if they are blank).